### PR TITLE
Bottom Navigation Styles

### PR DESCRIPTION
**Pull request by Juveniel**
targets: https://github.com/telerik/kendo-themes/issues/1949

### CODE REVIEW

**Review by silviyaboteva — APPROVED** [review]
Great job @Juveniel !

**Review by JoomFX — APPROVED** [review]

**Review by joneff — CHANGES_REQUESTED** [review]
* put link to the work item #1949
* the work items lacks description, design links, discussions etc.

After those are done, I can assess whether or not this item is implemented correctly

**Review by joneff — APPROVED** [review]

**Review by silviyaboteva — APPROVED** [review]

File: packages/default/scss/bottom-navigation/_theme.scss [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    @each $name, $color in $theme-colors {
+        .k-bottom-nav-solid.k-bottom-nav-#{$name} {
+            @include fill(
+                $color: true-mix( $color, contrast-wcag( $color ), 35%),
+                $bg: $color
+            );
+
+            .k-bottom-nav-item.k-state-focus,
+            .k-bottom-nav-item.k-state-focused,
+            .k-bottom-nav-item.k-state-focused:focus {
```
Comment (silviyaboteva) [inline packages/default/scss/bottom-navigation/_theme.scss:13]: may we skip the .k-state-focused class here and use only :focus?

File: packages/default/scss/bottom-navigation/_theme.scss [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    .k-bottom-nav-flat {
+        @include fill(
+            $bottom-nav-flat-text,
+            $bottom-nav-flat-bg,
+            $bottom-nav-flat-border
+        );
+
+        .k-bottom-nav-item.k-state-focus,
+        .k-bottom-nav-item.k-state-focused,
+        .k-bottom-nav-item.k-state-focused:focus {
```
Comment (silviyaboteva) [inline packages/default/scss/bottom-navigation/_theme.scss:35]: the same as above

File: tests/visual/bottom-nav.html [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            background-color: #F3F5F7;
+        }
+    </style>
+</head>
+<body>
+
+    <div id="test-area" class="grid">
+        <section>
+            <h4>Horizontal</h4>
+            <nav class="k-bottom-nav k-bottom-nav-primary k-bottom-nav-flat k-bottom-nav-border k-bottom-nav-item-flow-horizontal">
```
Comment (silviyaboteva) [inline tests/visual/bottom-nav.html:22]: Do we have a design update here as now I see that we have to expose `fill` prop in the API of the component that accepts `solid` and `flat` values. Also, we need to add it in the spec.

File: tests/visual/bottom-nav-colors.html [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+
+</head>
+<body>
+
+    <div id="test-area" class="grid">
+        <h2>Solid Colors</h2>
+        <div class="colors k-d-grid">
+            <section>
+                <h4>Primary ThemeColor</h4>
+                <nav class="k-bottom-nav k-bottom-nav-primary k-bottom-nav-solid k-bottom-nav-item-flow-vertical">
```
Comment (silviyaboteva) [inline tests/visual/bottom-nav-colors.html:31]: So, if it is `solid`, then the border is not applied, right?

File: tests/visual/bottom-nav-colors.html [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+                    </span>
+                </nav>
+            </section>
+        </div>
+
+        <h2>Flat Colors</h4>
+        <div class="colors colors-flat k-d-grid">
+            <section>
+                <h4>Primary ThemeColor</h4>
+                <nav class="k-bottom-nav k-bottom-nav-primary k-bottom-nav-flat k-bottom-nav-border k-bottom-nav-item-flow-vertical">
```
Comment (silviyaboteva) [inline tests/visual/bottom-nav-colors.html:232]: And if it is `flat`, then the border class is applied?

File: packages/default/scss/bottom-navigation/_layout.scss [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        outline: none;
+        -webkit-touch-callout: none;
+        -webkit-tap-highlight-color: $rgba-transparent;
+    }
+
+    .k-bottom-nav-border {
+        border-width: $bottom-nav-border-width;
+    }
+
+    .k-bottom-nav-shadow {
```
Comment (silviyaboteva) [inline packages/default/scss/bottom-navigation/_layout.scss:30]: I cannot find this class in the tests.

File: tests/visual/bottom-nav.html (hunk 3 above)
Comment (Juveniel) [reply tests/visual/bottom-nav.html:22]: The fill prop allows to differentiate between solid and flat theme variations and the change is synced with @tyanko . Design is not updated yet with latest changes.

Already added in the spec #1949

**Review by joneff — APPROVED** [review]

**Pull request merged**